dev_process.md

```diff
@@ -0,0 +1,86 @@
+4. Cleanup implementation
+5. Improve tests, then move back to 4 based on any found bugs
+6. [Fuzz test](https://book.getfoundry.sh/forge/fuzz-testing.html)
+  - Now that you are pretty confident in your implementation, throw a monkey at it. A good fuzz test should consider all valid inputs, and include as many state transition assertions as possible (think: is this function monotonically in/decreasing, should it be always less than something else,  etc.)
+7. Move back to step 4 if any bugs are found
+8. Integration test
+  - Your feature now likely does exactly what you think it does. In a complex system, that is not enough. Ideally you have tested how it affects the entire system as well. Invariant tests are coming soon to [Foundry](https://github.com/foundry-rs/foundry), which should help integration style tests, but make do with what you can with fuzz tests on a broader basis (not just for a single function)
+9. Move back to step 4 if any bugs are found
+10. Run [slither](https://github.com/crytic/slither)
+  - Analyze the output and return to step 4 if needed
+11. Cleanup documentation
+12. PR review
+  - The implementer is just the first line of defense. If you are a reviewer, confirm that the implementer followed the above principals (test-per-state-transition, test-per-revert, fuzz test, and integration test)
+  - Review the documentation and ensure the implementation matches the documented behavior. If it does not, touch base with the implementer and confirm which needs to be updated.
+13. Deployment script
+  - Write the deployment script
+  - Most likely Foundry has scripting ready when you are reading this. Check out this [PR](https://github.com/foundry-rs/foundry/pull/1208)
+14. Deployment test
+  - Ensure deployment goes exactly as planned by writing a test testing *every state transition* and make sure no changes unexpectedly happen. One way to accomplish this is the `record` cheatcode. Create a list of your entire protocol's addresses, call record, perform the upgrade, call `accesses` for each address of your protocol. Ensure there are no slots/addresses that unexpectedly changed.
+15. Audit
+  - Consider if this feature/contract needs an audit. Always lean towards being safe rather than sorry.
+  - Have the complexity & code size inform if and who should audit your contracts
+  - At Nascent, we have an internal tier list of quality of auditors. You should check with other developers about who are good auditors and who aren't. Some auditors are just there to check a box for a protocol, others actually care about finding vulnerabilities.
+16. Implement audit fixes
+17. Setup monitoring service
+  - Have an internal tool that monitors important aspects of your system.
+  - Use tools like [Check the Chain](https://github.com/fei-protocol/checkthechain), [Grafana](https://grafana.com/), or use an off-the-shelf monitoring tool like [tenderly](https://tenderly.co/alerting)
+18. Deploy
```

Comment:
Any deployment tools to recommend or is it super obvious and straightforward?

Reply:
See:
```
13. Deployment script
    - Write the deployment script
    - Most likely Foundry has scripting ready when you are reading this. Check out this [PR](https://github.com/foundry-rs/foundry/pull/1208)
```

Deploy here is just meant to convey "Run Deployment Script". Can reword if you think I should